Purchases/FoundationExtensions/Error+Extensions.swift

```diff
@@ -56,4 +60,12 @@ extension NSError {
         return userInfo[Backend.RCAttributeErrorsKey as String] as? [String: String]
     }
 
+    func clonedErrorWithMergedUserInfo(newUserInfoItems: [String: Any]) -> NSError {
+        var currentUserInfo = userInfo
+        newUserInfoItems.forEach { (key, value) in
```

Comment:
is there no `merge` function or anything for dictionaries?

Reply:
Done!

Reply:
removed this code as I went with a custom error type for DNSChecker